tools/c7n_gcp/c7n_gcp/resources/spanner.py

```diff
@@ -44,6 +46,87 @@ def get_label_params(resource, all_labels):
                         'field_mask': ', '.join(['labels'])}}
 
 
+@resources.register('spanner-instance-backup')
+class SpannerInstanceBackup(ChildResourceManager):
+    """GC resource: https://cloud.google.com/spanner/docs/reference/rest/v1/projects.instances.backups"""
+    class resource_type(ChildTypeInfo):
+        service = 'spanner'
+        version = 'v1'
+        component = 'projects.instances.backups'
+        enum_spec = ('list', 'backups[]', None)
+        scope = 'parent'
+        name = id = 'backups'
+        parent_spec = {
+            'resource': 'spanner-instance',
+            'child_enum_params': {
+                ('instances', 'parent')},
+            'use_child_query': True,
+        }
+        default_report_fields = ['name', 'expireTime']
+        permissions = ('spanner.backups.list',)
+        asset_type = 'spanner.googleapis.com/Backup'
+
+    def _get_child_enum_args(self, parent_instance):
+        return {
+            'parent': 'projects/{}/instances/{}'.format(
+                local_session(self.session_factory).get_default_project(),
+                parent_instance['displayName'],
+            )
+        }
+
+
+@SpannerInstanceBackup.filter_registry.register('time-range')
+class SpannerInstanceBackupTimeRangeFilter(Filter):
+    """Filters bigtable instance clusters backups based on a time range
+
+    .. code-block:: yaml
+
+        policies:
+          - name: bigtable_backup_expiration_time_30_days
+            description: |
+              Cloud Bigtable backup expiration time is 29 days or less
+            resource: gcp.bigtable-instance-cluster-backup
+            filters:
+              - type: time-range
+                value: 29
+    """
+    schema = type_schema('time-range',
+                         value={'$ref': '#/definitions/filters_common/value'})
+    permissions = ('bigtable.backups.list',)
+
+    datetime1_pattern = "%Y-%m-%dT%H:%M:%S.%fZ"
+    datetime2_pattern = "%Y-%m-%dT%H:%M:%S"
+
+    create_time_field_name = 'createTime'
+    expire_time_field_name = 'expireTime'
+
+    def process(self, resources, event=None):
+        filtered_resources = []
+        value = self.data.get('value')
+
+        for resource in resources:
+            create_time_pattern = self.datetime1_pattern
```

Comment:
why is this logic being duplicated?

Reply:
added a new common filter and re-used it from bigtable and spanner